replica_set_checker.go

```diff
@@ -0,0 +1,102 @@
+package dvara
+
+import "errors"
+
+type ReplicaSetChecker struct {
+	Log        Logger
+	replicaSet *ReplicaSet
+}
+
+type ReplicaSetComparison struct {
+	// Extra members that are in this state, but not in new state
+	ExtraMembers map[string]*Proxy
+	// Missing members that aren't in this state, but are in new
+	MissingMembers map[string]*Proxy
+}
+
+func (checker *ReplicaSetChecker) Check() error {
+	addrs := checker.replicaSet.lastState.Addrs()
+	r, err := checker.replicaSet.ReplicaSetStateCreator.FromAddrs(checker.replicaSet.Username, checker.replicaSet.Password, addrs, checker.replicaSet.Name)
```

Comment:
Cool. How about adding two metrics:
 To FromAddrs methods, to see how long it takes, and if it failed.
 To this specific line, to see if checker failed. Since if we return here we failed.